include/aws/s3/private/s3_meta_request_impl.h

```diff
@@ -128,6 +128,7 @@ struct aws_s3_meta_request_vtable {
  */
 struct aws_s3_meta_request {
     struct aws_allocator *allocator;
+    struct aws_s3_buffer_pool *buffer_pool;
```

Comment:
is this just an ergonomic thing? I'd prefer we be clear about ownership and just do `request->client->buffer_pool`?
```suggestion
```

Reply:
debatable/trivial: still think `request->client->buffer_pool` is better than redundant pointer (extra variable makes ownership unclear, and it's one more thing that might get out of sync, or forget to set)

but also whatever